### PR TITLE
Fix RabbitMQ regression in #9383

### LINKED ISSUE #9383: RabbitMQ Input Plugin v1.19.0 unmarshal errors
### Relevant telegraf.conf:

```
[[inputs.rabbitmq]]
  ## Management Plugin url. (default: http://localhost:15672)
   url = "http://1.2.3.4:15672"
   username = "username"
   password = "password"
```

### System info:

Windows Server 2019
Telegraf v1.19.0
RabbitMQ 3.8.14
Erlang 23.2.7

### Steps to reproduce:

1. Configure inputs.rabbitmq
2. Run a test collection 
telegraf.exe --config "C:\Program Files\Telegraf\telegraf.conf" --test --input-filter rabbitmq --debug

### Expected behavior:

Collection should complete without errors.

### Actual behavior:
There are unmarshal errors within the collection 

2021-06-16T16:17:01Z E! [inputs.rabbitmq] Error in plugin: json: cannot unmarshal object into Go value of type []rabbitmq.FederationLink
2021-06-16T16:17:01Z E! [inputs.rabbitmq] Error in plugin: json: cannot unmarshal number 0.0 into Go struct field Node.io_read_avg_time of type int64
2021-06-16T16:17:01Z E! [telegraf] Error running agent: input plugins recorded 2 errors

### Additional info:

Telegraf v1.18.1 on the same system does not show the unmarshal errors.

### PR DESCRIPTION
**Pull request by srebhan**
### Required for all PRs:

- [X] Updated associated README.md.
- [X] Wrote appropriate unit tests.

resolves #9383 

This resolves the wrong types for `_avg` metrics. Those errors went unnoticed before checking the return types (#8992) and returned invalid values (all zero) if the server responded with as floating-point formatted number by chance.
This PR furthermore tries to handle missing plugins (e.g. `federation`) more gracefully providing the server sent error to the user. To avoid those errors altogether, a (backward-compatible) metric-filter is introduced such that the user can disable erroneous metrics.

### CODE REVIEW

**Review by akrantz01 — CHANGES_REQUESTED** [review]
Overall LGTM. I just left some comments on some of the type conversion for `Memory.Total`.

File: plugins/inputs/rabbitmq/rabbitmq.go [hunk 1]
```diff
@@ -533,7 +589,31 @@ func gatherNodes(r *RabbitMQ, acc telegraf.Accumulator) {
 				fields["mem_other_system"] = memory.Memory.OtherSystem
 				fields["mem_allocated_unused"] = memory.Memory.AllocatedUnused
 				fields["mem_reserved_unallocated"] = memory.Memory.ReservedUnallocated
-				fields["mem_total"] = memory.Memory.Total
+				switch v := memory.Memory.Total.(type) {
+				case int64:
+					fields["mem_total"] = v
+				case float64:
+					fields["mem_total"] = int64(v)
+				case map[string]interface{}:
```
[the remark this answers is not in the document]
Comment (srebhan) [reply plugins/inputs/rabbitmq/rabbitmq.go:597]: Unfortunately, this does not work as the JSON decoder will return a `map[string]interface{}`.

File: plugins/inputs/rabbitmq/rabbitmq.go [hunk 2]
```diff
@@ -533,7 +589,31 @@ func gatherNodes(r *RabbitMQ, acc telegraf.Accumulator) {
...
+					var total interface{}
+					if x, found := v["rss"]; found {
+						total = x
+					} else if x, found := v["allocated"]; found {
+						total = x
+					} else if x, found := v["erlang"]; found {
+						total = x
+					} else {
+						acc.AddError(fmt.Errorf("no known memory estimation in %v", v))
```
[the remark this answers is not in the document]
Comment (srebhan) [reply plugins/inputs/rabbitmq/rabbitmq.go]: Done. Simplified the whole estimator thing, so I think we avoid this issue now with less code. :-)

File: plugins/inputs/rabbitmq/rabbitmq.go [hunk 3]
```diff
@@ -533,7 +589,31 @@ func gatherNodes(r *RabbitMQ, acc telegraf.Accumulator) {
 				fields["mem_other_system"] = memory.Memory.OtherSystem
 				fields["mem_allocated_unused"] = memory.Memory.AllocatedUnused
 				fields["mem_reserved_unallocated"] = memory.Memory.ReservedUnallocated
-				fields["mem_total"] = memory.Memory.Total
+				switch v := memory.Memory.Total.(type) {
+				case int64:
+					fields["mem_total"] = v
```
[the remark this answers is not in the document]
Comment (srebhan) [reply plugins/inputs/rabbitmq/rabbitmq.go]: I take your work on it. ;-) Changed.

**Comment by srebhan** [comment]
@akrantz01 thanks for review. I changed the code according to your suggestions except for the `map[string]interface{}` part, which does not work otherwise.

**Review by akrantz01 — APPROVED** [review]

**Review by sspaink — APPROVED** [review]

**Pull request merged**